src/serializers/type_serializers/definitions.rs

```diff
@@ -89,10 +86,10 @@ impl TypeSerializer for DefinitionRefSerializer {
         exclude: Option<&PyAny>,
         extra: &Extra,
     ) -> Result<S::Ok, S::Error> {
-        let value_id = extra.rec_guard.add(value).map_err(py_err_se_err)?;
-        let comb_serializer = unsafe { extra.slots.get_unchecked(self.serializer_id) };
```

Comment:
I spent a long time debugging these unsafes. Totally not worth it: unsafe only makes sense when the thing isn't possible or is very cumbersome to do without unsafe. Here it's the same number of characters, the same ease of comprehension and all it costs us is a bounds check (which is great because it gives you a traceback instead of a `bus error`). The algorithm is also super complex and this is an essential part of it. Let's _not_ use unsafe in a critical piece of the iterative flattening of a recursion.

Reply:
ok.